### PR TITLE
[cmake]: Second try at making swift-api-digester work correctly.

### PR DESCRIPTION
**Pull request by erg**
Running ninja check-swift didn't rebuild swift-api-digester.

Previous: https://github.com/apple/swift/pull/5505
and: https://github.com/apple/swift/pull/5509

rdar://problem/28953348

### CODE REVIEW

**Comment by erg** [comment]
@swift-ci Please smoke test

**Comment by jrose-apple** [comment]
The problem is that swift-api-digester is only built on Apple platforms.

(I mean, maybe we want to change that—cc @nkcsgexi—but for now I don't think we can just turn it on and expect things to work.)

**Comment by nkcsgexi** [comment]
Can we exclusively build swift-api-digester on Apple platforms for now?

**Comment by nkcsgexi** [comment]
@swift-ci Please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci Please smoke test

**Comment by nkcsgexi** [comment]
swift-api-digester now builds on linux but the smoke test failed for other reasons. Try again..
@swift-ci please smoke test

File: tools/CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,7 @@ add_swift_tool_subdirectory(swift-demangle)
 add_swift_tool_subdirectory(lldb-moduleimport-test)
 add_swift_tool_subdirectory(sil-extract)
 add_swift_tool_subdirectory(swift-llvm-opt)
+add_swift_tool_subdirectory(swift-api-digester)
 
 if(SWIFT_HOST_VARIANT STREQUAL "macosx")
   # Only build Darwin-specific tools when deploying to OS X.
```
Comment (jrose-apple) [inline tools/CMakeLists.txt]: If we are trying to build swift-api-digester for all platforms, please remove this section.

Comment (erg) [inline tools/CMakeLists.txt]: Sorry, I missed this section. I see how my patch could be fixed for just OSX support, but is this something we want for Linux too?

**Comment by erg** [comment]
Cleaned up the OSX special casing for swift-api-digester.

**Comment by erg** [comment]
@swift-ci Please smoke test

**Pull request merged**